### PR TITLE
update bitcoin to 0.31.2

### PR DESCRIPTION
**Pull request by getong**
#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

wait for 
- [ ] https://github.com/bitcoindevkit/rust-hwi/pull/91
- [ ] https://github.com/bitcoindevkit/rust-esplora-client/pull/60

### CODE REVIEW

File: .github/workflows/cont_integration.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
         rust:
           - version: stable
             clippy: true
-          - version: 1.63.0 # MSRV
```
Comment (danielabrozzoni) [inline .github/workflows/cont_integration.yml:15]: Why are you updating the project MSRV?

**Review by danielabrozzoni — COMMENTED** [review]
Hey, I still haven't looked at the code in depth, but I have a few comments:
- You are upgrading the project MSRV from 1.63 to 1.70 - that's not a change we take lightly, is there any reason behind it? Has this been discussed with the team?
- Commits should be atomic and hygienic, and the message should explain what you're doing. You currently have multiple commits doing the same thing, which should be squashed together. I suppose this PR should have one commit updating to rust-bitcoin 0.31 + more commits if you're doing something else (updating the MSRV, for example, would need its own commit). More info here: https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#committing-patches or here: https://blog.lanesawyer.dev/17501/git-hygiene
- The CI is currently failing, can you look into that please?

**Comment by getong** [comment]
I have not finished the upgrade code, some code test are not passed.
Need more time.

**Comment by danielabrozzoni** [comment]
No problem! I suggest you keep this PR in draft in the meantime, then :)

**Pull request closed without merging**